spec/commands/remove/group_spec.rb

```diff
@@ -40,29 +40,42 @@
   end
 
   let :config { Metalware::Config.new }
+  let :loader { Metalware::Validator::Loader.new(config) }
+  let :cache { loader.groups_cache[:primary_groups] }
+
+  let :initial_files { answer_files }
+  let :deleted_files { initial_files - answer_files }
 
   before :each do
     SpecUtils.mock_validate_genders_success(self)
     SpecUtils.use_mock_genders(self, genders_file: 'setup1/genders')
+    filesystem.test { initial_files }
+  end
+
+  def answer_files
+    Dir[File.join(config.answer_files_path, '**/*.yaml')]
   end
 
-  def generate_node_list(prefix, max_index: 10)
-    [*1..max_index].map { |idx| "#{prefix}#{'0' if idx < 10}#{idx}" }
+  def expected_deleted_files(group)
+    Metalware::NodeattrInterface.nodes_in_primary_group(group)
+      .map { |node| "nodes/#{node}.yaml" }
+      .unshift(["groups/#{group}.yaml"])
+      .map { |f| File.join(config.answer_files_path, f) }
   end
 
-  def run_remove_group(primary_group, primary_nodes, &test_block)
+  def test_remove_group(group)
     filesystem.test do |_fs|
-      answer_check = RSpecRemoveGroup::AnswerFileChecker
-                     .new(self, config, primary_group, primary_nodes)
-      Metalware::Commands::Remove::Group.new([primary_group], OpenStruct.new)
-      answer_check.check
-      yield if test_block
+      Metalware::Commands::Remove::Group.new([group], OpenStruct.new)
+      expect(expected_deleted_files(group)).to include(*deleted_files)
```

Comment:
Thanks for making these changes, this looks clearer to me now. Can this line use `eq` rather than `include`? As these should be exactly the same shouldn't they?